src/Api/FormatsProvider.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Api;
+
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Util\RequestAttributesExtractor;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * {@inheritdoc}
+ *
+ * @author Anthony GRASSIOT <antograssiot@free.fr>
+ */
+final class FormatsProvider implements FormatsProviderInterface
+{
+    private $configuredFormats;
+    private $formats;
+    private $resourceMetadataFactory;
+
+    public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFactory, array $formats)
+    {
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->configuredFormats = $formats;
+        $this->formats = [];
+    }
+
+    /**
+     * Finds formats from a Resource class.
+     */
+    public function fromRequest(Request $request): array
+    {
+        if ($this->formats) {
```

Comment:
It looks buggy: if a different format can be configured per resource, then you must test for `$this->formats[$resourceClass]` or the bad formats will be returned (for instance the ones of the previously requested resource class).

Reply:
I'm not storing key/values just the format for one operation right now